Sprint-3/revise/implement/password-validator.test.js

```diff
@@ -14,3 +17,58 @@ To be valid, a password must:
 
 You must breakdown this problem in order to solve it. Find one test case first and get that working
 */
+function passwordValidation(pass) {
+  let bannedPasswords = ["ydt635RERE!!", "hdud738#&sg", "bxv735#$ls0HG"];
+  if (
+    /[A-Z]/.test(pass) &&
+    /[a-z]/.test(pass) &&
+    pass.length >= 5 &&
+    /[0-9]/.test(pass) &&
+    /[!#$%.*&]/.test(pass) &&
+    bannedPasswords.every((password) => password !== pass) &&
+    !/[" ")(£`¬|\\]/.test(pass)
+  )
+    return "Valid Password!";
+  else return "Invalid Password";
+}
+it("should allow valid passwords", () => {
+  expect(passwordValidation("gah45SA!")).toBe("Valid Password!");
+  expect(passwordValidation("gah45SA#")).toBe("Valid Password!");
+});
+it("should not allow invalid password", () => {
+  expect(passwordValidation("hssssgdwu")).toBe("Invalid Password");
+  expect(passwordValidation("DDTTUOLH")).toBe("Invalid Password");
+  expect(passwordValidation("$.*!$£")).toBe("Invalid Password");
+  expect(passwordValidation("165875584")).toBe("Invalid Password");
+  expect(passwordValidation("Lo 5#j")).toBe("Invalid Password");
+  expect(passwordValidation("L2m()!")).toBe("Invalid Password");
+  expect(passwordValidation("1`\\UYgs")).toBe("Invalid Password");
+});
+it("should not be less than 5 digits", () => {
+  expect(passwordValidation("hs3Q")).toBe("Invalid Password");
```

Comment:
"h$3Q" would be a better test case than "hs3Q" because the former can pass all other checks except the one that checks for password length.

Reply:
oh Yes! That's right! amended.
Thanks